src/jvmTest/kotlin/com/charleskorn/kaml/JvmYamlReadingTest.kt

```diff
@@ -42,5 +44,18 @@ class JvmYamlReadingTest : DescribeSpec({
                 result shouldBe 123
             }
         }
+
+        describe("reading an empty yaml without throwing an error") {
+            val input = ""
+            val bytes = ByteArrayInputStream(input.toByteArray(Charsets.UTF_8))
+
+            it("expect throwing an error because it's an empty document") {
+                shouldThrowExactly<EmptyYamlDocumentException> { Yaml.default.decodeFromStream<String>(bytes) }
```

Comment:
It would be good to add tests to cover a couple of other scenarios as well. For example, reading lists, maps and Kotlin classes.